wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java

```diff
@@ -249,61 +273,69 @@ public void doProcessing(IProgressMonitor monitor) {
 			if (pathMap.report_type.endsWith("_post")) {
 				calculate_dts = true;
 			}
-			DataReference refBase = Utils.getReference(dssGroupBase,
-					pathMap.pathBase, calculate_dts, pathnameMaps, 1);
-			DataReference refAlt = Utils.getReference(dssGroupAlt,
-					pathMap.pathAlt, calculate_dts, pathnameMaps, 2);
-			if (refBase == null || refAlt == null) {
+			tscBase = null;
+			tscAlt = null;
+			if (!pathMap.pathAlt.equalsIgnoreCase("ignore")) {
+				searchPathBase = Utils.substitutePartIntoPath(pathMap.pathBase, baseAPart, 1);
+				searchPathBase = Utils.substitutePartIntoPath(searchPathBase, baseFPart, 6);
+				tscBase = Utils.getTSContainer(htsBase, searchPathBase, calculate_dts);
+			}
+			if (!pathMap.pathAlt.equalsIgnoreCase("ignore")) {
+				searchPathAlt = Utils.substitutePartIntoPath(pathMap.pathAlt, altAPart, 1);
+				searchPathAlt = Utils.substitutePartIntoPath(searchPathAlt, altFPart, 6);
+				tscAlt = Utils.getTSContainer(htsAlt, searchPathAlt, calculate_dts);
+			}			if (tscBase == null || tscAlt == null) {
 				continue;
 			}
 			String[] series_name = new String[] { scalars.get("NAME_ALT"), scalars.get("NAME_BASE") };
 			if (pathMap.units.equalsIgnoreCase("CFS2TAF")) {
-				TSMath.cfs2taf((RegularTimeSeries) refBase.getData());
-				TSMath.cfs2taf((RegularTimeSeries) refAlt.getData());
-			} else if (pathMap.units.equalsIgnoreCase("TAF2CFS") || pathMap.units.equalsIgnoreCase("CFS")) {
-				TSMath.taf2cfs((RegularTimeSeries) refBase.getData());
-				TSMath.taf2cfs((RegularTimeSeries) refAlt.getData());
+				tscBase=Utils.cfs2taf(tscBase);
+				tscAlt=Utils.cfs2taf(tscAlt);
+			} else if (pathMap.units.equalsIgnoreCase("TAF2CFS")) {
+				tscBase=Utils.taf2cfs(tscBase);
+				tscAlt=Utils.taf2cfs(tscAlt);
 			}
-			String data_units = Utils.getUnits(refBase, refAlt);
-			String data_type = Utils.getType(refBase, refAlt);
+			String data_units = tscBase.units;
+			String data_type = tscBase.type;
```

Comment:
TimeSeriesContainer.type is going to give you things like "PER-AVER" or "INST-VAL". You probably want to set your data_type from tscBase.parameter, which is going to be the C-part of the DSS path

Reply:
per @hxiedwr "Right now, we don’t have console outputs but haven’t got any complaints in the past. So I think this is not urgent. "

Reply:
Sorry -- that last comment belongs in the other location.

Reply:
Per @hxiedwr -- "The same goes to data_type, which is not used to generate tables or charts in the Report Tool."
If this won't cause problems with the labels of the axes, then it's not significant.